upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -132,15 +121,39 @@ func (e *AutoscalingGroup) Find(c *fi.Context) (*AutoscalingGroup, error) {
 	if len(g.Tags) != 0 {
 		actual.Tags = make(map[string]string)
 		for _, tag := range g.Tags {
-			actual.Tags[*tag.Key] = *tag.Value
+			actual.Tags[fi.StringValue(tag.Key)] = fi.StringValue(tag.Value)
 		}
 	}
 
-	if fi.StringValue(g.LaunchConfigurationName) == "" {
-		glog.Warningf("autoscaling Group %q had no LaunchConfiguration", fi.StringValue(g.AutoScalingGroupName))
-	} else {
+	if g.LaunchConfigurationName != nil {
 		actual.LaunchConfiguration = &LaunchConfiguration{ID: g.LaunchConfigurationName}
 	}
+	if g.LaunchTemplate != nil {
+		actual.LaunchTemplate = &LaunchTemplate{ID: g.LaunchTemplate.LaunchTemplateName}
+	}
+
+	if g.MixedInstancesPolicy != nil {
+		mp := g.MixedInstancesPolicy
+		if mp.InstancesDistribution != nil {
+			mpd := mp.InstancesDistribution
+			actual.MixedOnDemandAboveBase = mpd.OnDemandPercentageAboveBaseCapacity
+			actual.MixedOnDemandAllocationStrategy = mpd.OnDemandAllocationStrategy
+			actual.MixedOnDemandBase = mpd.OnDemandBaseCapacity
+			actual.MixedSpotAllocationStrategy = mpd.SpotAllocationStrategy
+			actual.MixedSpotAllocationStrategy = mpd.SpotAllocationStrategy
+			actual.MixedSpotInstancePools = mpd.SpotInstancePools
```

Comment:
Would this value potentially be overridden / change, if the instance types in `MixedInstanceOverrides` exceeds it?

Reply:
I think the wording confused me, this is related to Spot Diversity: `Diversify Spot Instances across your X lowest priced instance types per Availability Zone`